frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -881,4 +929,75 @@ mod tests {
 			assert!(matches!(call, OuterCall::MultiPhase(Call::submit_unsigned(_, _))));
 		})
 	}
+
+	#[test]
+	fn trim_compact_length_does_not_modify_when_short_enough() {
+		let (mut ext, _) = ExtBuilder::default().build_offchainify(0);
+		ext.execute_with(|| {
+			roll_to(25);
+
+			let RoundSnapshot { voters, ..} =
+				MultiPhase::snapshot().unwrap();
+
+			let RawSolution { mut compact, .. } = raw_solution();
+			let encoded_len = compact.encode().len() as u32;
+			let compact_clone = compact.clone();
+
+			compact = MultiPhase::trim_compact_length(
+				encoded_len,
+				compact,
+				voter_index_fn_linear::<Runtime>(&voters),
+			).unwrap();
+
+			assert_eq!(compact, compact_clone);
+		});
+	}
+
+	#[test]
+	fn trim_compact_length_modifies_when_too_long() {
+		let (mut ext, _) = ExtBuilder::default().build_offchainify(0);
+		ext.execute_with(|| {
+			roll_to(25);
+
+			let RoundSnapshot { voters, ..} =
+				MultiPhase::snapshot().unwrap();
+
+			let RawSolution { mut compact, .. } = raw_solution();
+			let encoded_len = compact.encode().len() as u32;
+			let compact_clone = compact.clone();
+
+			compact = MultiPhase::trim_compact_length(
+				encoded_len - 1,
+				compact,
+				voter_index_fn_linear::<Runtime>(&voters),
+			).unwrap();
+
+			assert_ne!(compact, compact_clone);
+			assert!((compact.encode().len() as u32) < encoded_len);
```

Comment:
```suggestion
			assert!(compact.encoded_size() as u32 < encoded_len);
```
`

Reply:
Can't get rid of the parens, unfortunately:

```
error: `<` is interpreted as a start of generic arguments for `u32`, not a comparison
   --> frame/election-provider-multi-phase/src/unsigned.rs:986:42
    |
986 |             assert!(compact.encoded_size() as u32 < encoded_len);
    |                     ----------------------------- ^ ----------- interpreted as generic arguments
    |                     |                             |
    |                     |                             not interpreted as comparison
    |                     help: try comparing the cast value: `(compact.encoded_size() as u32)`
```

56514573b implements the rest of the suggestion, though.